src/sentry/rules/processing/processor.py

```diff
@@ -272,9 +299,13 @@ def apply_rule(self, rule: Rule, status: GroupRuleStatus) -> None:
 
         state = self.get_state()
         condition_list, filter_list = split_conditions_and_filters(rule.data.get("conditions", ()))
-
-        # Sort `condition_list` so that most expensive conditions run last.
-        condition_list.sort(key=lambda condition: is_condition_slow(condition))
+        slow_conditions = None
+        if features.has("organizations:process-slow-alerts", self.project.organization):
+            fast_conditions, slow_conditions = self.group_conditions_by_speed(condition_list)
+            condition_list = fast_conditions
```

Comment:
🤔 hmmm can we name this a different variable?

feels a bit smelly using `conditions_list` to derive `fast_conditions`, then re-setting conditions list to _be_ fast_conditions

Reply:
I see so it looks like we're either determining whether to run slow conditions last, or skip slow conditions altogether?

Maybe we can _always_ group conditions by speed, and if they _don't_ have the slow conditions flag, we append the slow conditions to the end of the list?

Reply:
yeah I think that would be a bit clearer as well cuz in both cases we're sorting by condition speed

Reply:
not sure why but I couldn't get this to actually work, the condition list looks the same when printed but a test fails 🤔